File: 삽입정렬.py
# 삽입정렬

n = int(input())

list = []

for _ in range(1, n+1):
	list.append(int(input()))

for i in range(1, n):
    for j in range(i, 0, -1):
        if (list[j] < list[j-1]):
            tmp = list[j-1]
            list[j-1] = list[j]
            list[j] = tmp

        else:
            break

print(list)

File: 삽입정렬.cpp
#include <stdio.h>
int main(){
    int n,i,j,k,d[101], tmp;
    scanf("%d",&n);
    for(i=0;i<n;i++){
        scanf("%d",&d[i]);
    }
    for(i=1;i<n;i++){
        for(j=i;j>0;j--){
            if(d[j]<d[j-1]){
                tmp = d[j];
                d[j] = d[j-1];
                d[j-1] = tmp;
            }
            else{
                break;
            }

        }
    }

    for(i=0;i<n;i++){
        printf("%d ", d[i]);
    }


}
